contrib/android/include/android_lf.h

```diff
@@ -0,0 +1,39 @@
+/* 
+ * Macros for file64 functions
+ *
+ * Android does not support the macro _FILE_OFFSET_BITS=64
+ * As of android-21 it does however support many file64 functions
+*/
+
+/* EXPLICITLY NO HEADER GUARD */

```

Comment:
To explain this:

There isn't really a consenus on the order of `#include`. Some files have system api header first and some files libarchive headers first. An example might look like this

```
#include "archive.h" //includes android_lf.h and defines stat as stat64
#include <sys/stat.h> //defines struct stat --> error stat is already defined ^
#include "archive_entry.h" //includes android_lf.h
```

So to fix this issue I'm only redefining stat if it's already defined. Because it doesn't have header guards, the second time it's included, it will get defined in the example above.